### PR TITLE
Scripting: added set_rpm_scale ESC telem API

### PR DESCRIPTION
**Pull request by tridge**
This allows for vehicles with different numbers of poles per motor by running a lua script which adds a scale factor for ESC RPMs

### CODE REVIEW

File: libraries/AP_ESC_Telem/AP_ESC_Telem.h [hunk 1]
```diff
@@ -108,6 +115,12 @@ class AP_ESC_Telem {
     uint32_t _last_rpm_log_us[ESC_TELEM_MAX_ESCS];
     uint8_t next_idx;
 
+#if AP_SCRIPTING_ENABLED
+    // allow for scaling of RPMs via lua scripts
```
Comment (andyp1per) [inline libraries/AP_ESC_Telem/AP_ESC_Telem.h:119]: This is always related to the number of poles via ```erpm * (200 / _bdshot.motor_poles)```. I think a much better change is to record whether an ESC is recording in rpm or erpm and then make the appropriate calculation.

**Review by andyp1per — COMMENTED** [review]
This is definitely a hack, will need to think about it a bit more.

File: libraries/AP_ESC_Telem/AP_ESC_Telem.h (hunk 1 above)
Comment (tridge) [reply libraries/AP_ESC_Telem/AP_ESC_Telem.h:119]: how would we record that, noting that it could be different for each ESC

**Review by andyp1per — APPROVED** [review]

**Review by rmackay9 — APPROVED** [review]

**Comment by peterbarker** [comment]
Just DevCallTopic in case it doesn't get merged before then; not sure what the recent pushes changed after approval was given.

**Comment by IamPete1** [comment]
Fixes https://github.com/ArduPilot/ardupilot/issues/9982

**Review by andyp1per — APPROVED** [review]
Still looks right to me. I think probably you should test the various CAN/IOMCU/Regular scenarios to make sure that each of these is addressable via the mask.

**Pull request merged**